build-support/common.sh

```diff
@@ -68,3 +68,22 @@ function git_merge_base() {
   # This prints the tracking branch if set and otherwise falls back to local "master".
   git rev-parse --symbolic-full-name --abbrev-ref HEAD@\{upstream\} 2>/dev/null || echo 'master'
 }
+
+function determine_python() {
+  if [[ -n "${PY}" ]]; then
+    echo "${PY}"
+    return 0
+  fi
+  for version in '3.7' '3.8'; do
+    local interpreter_path
+    interpreter_path="$(command -v "python${version}")"
+    if [[ -z "${interpreter_path}" ]]; then
+      continue
+    fi
+    # Check if the Python version is installed via Pyenv but not activated.
+    if [[ "$("${interpreter_path}" --version 2>&1 > /dev/null)" == "pyenv: python${version}"* ]]; then
```

Comment:
I don't follow this logic after playing with pyenv here locally. It would seem more straight forward to use the pyenv equivalent of command -v:
```
$ pyenv local 3.7.6
$ pyenv which python3.7
/home/jsirois/.pyenv/versions/3.7.6/bin/python3.7
$ pyenv which python3.8 2>/dev/null
$
```

Reply:
Ok, I get the logic now. That's pretty hacky but it works fine.